openfisca_uk/variables/hmrc/stamp_duty.py

```diff
@@ -0,0 +1,111 @@
+from openfisca_uk.model_api import *
+
+
+class stamp_duty_on_residential_property(Variable):
+    label = "Stamp Duty on residential property"
+    documentation = (
+        "Tax charge from purchase or rental of residential property"
+    )
+    entity = Household
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(household, period, parameters):
+        stamp_duty = parameters(period).hmrc.stamp_duty
+        # Tax on main-home purchases
+        price_limit = stamp_duty.residential.purchase.main.first.max
+        price = household("main_residential_property_purchased", period)
+        residential_purchase_qualifies_as_first_buy = household(
+            "main_residential_property_purchased_is_first_home", period
+        ) & (price < price_limit)
+        main_residential_purchase_tax = where(
+            residential_purchase_qualifies_as_first_buy,
+            stamp_duty.residential.purchase.main.first.rate.calc(price),
+            stamp_duty.residential.purchase.main.subsequent.calc(price),
+        )
+        # Tax on second-home purchases
+        second_home_price = household(
+            "additional_residential_property_purchased", period
+        )
+        price = where(
+            second_home_price < stamp_duty.residential.purchase.additional.min,
+            0,
+            second_home_price,
+        )
+        additional_residential_purchase_tax = (
+            stamp_duty.residential.purchase.additional.rate.calc(price)
+        )
+        # Tax on residential rents
+        cumulative_rent = household("cumulative_residential_rent", period)
+        rent = household("rent", period)
+        residential_rent_tax = stamp_duty.residential.rent.calc(
```

Comment:
I haven't seen this notation before - parameters can call `calc`?

Reply:
It's the ParameterScale interface - used in rates, etc. in tax calculations too